Userland/Libraries/LibSQL/AST/Lexer.cpp

```diff
@@ -177,91 +186,137 @@ bool Lexer::consume_whitespace_and_comments()
 }
 
-bool Lexer::consume_string_literal()
+bool Lexer::consume_string_literal(StringBuilder& current_token)
 {
     // https://sqlite.org/lang_expr.html - See "3. Literal Values (Constants)"
     bool is_valid_string_literal = true;
+
+    // Skip the opening single quote:
     consume();
 
-    while (!is_eof() && !is_string_literal_end())
-        consume();
+    while (!is_eof() && !is_string_literal_end()) {
+        // If both the current character and the next one are single quotes,
+        // consume one single quote into the current token, and drop the
+        // other one on the floor:
+        if (match('\'', '\''))
+            consume();
+        consume(&current_token);
+    }
 
     if (is_eof())
         is_valid_string_literal = false;
+    // Drop the closing quote on the floor:
     consume();
 
     return is_valid_string_literal;
 }
 
-bool Lexer::consume_blob_literal()
+bool Lexer::consume_quoted_identifier(StringBuilder& current_token)
 {
-    // https://sqlite.org/lang_expr.html - See "3. Literal Values (Constants)"
+    // I have not found a reference to the syntax for identifiers in the
+    // SQLite documentation, but PostgreSQL has this:
+    // https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS
+    bool is_valid_identifier = true;
+
+    // Skip the opening double quote:
     consume();
-    return consume_string_literal();
+
+    while (!is_eof() && !is_quoted_identifier_end()) {
+        // If both the current character and the next one are double quotes,
+        // consume one single quote into the current token, and drop the
+        // other one on the floor:
+        if (match('"', '"'))
+            consume();
+        consume(&current_token);
+    }
+
+    if (is_eof())
+        is_valid_identifier = false;
+    // Drop the closing double quote on the floor:
+    consume();
+
+    return is_valid_identifier;
 }
 
-bool Lexer::consume_exponent()
+bool Lexer::consume_blob_literal(StringBuilder& current_token)
 {
+    // https://sqlite.org/lang_expr.html - See "3. Literal Values (Constants)"
+
+    // Skip starting 'X'/'x' character:
     consume();
+
+    if (!consume_string_literal(current_token))
+        return false;
+    for (auto ix = 0u; ix < current_token.length(); ix++) {
+        if (!isxdigit(current_token.string_view()[ix]))
+            return false;
+    }
+    return true;
+}
+
+bool Lexer::consume_exponent(StringBuilder& current_token)
+{
+    consume(&current_token);
     if (m_current_char == '-' || m_current_char == '+')
-        consume();
+        consume(&current_token);
 
     if (!isdigit(m_current_char))
         return false;
 
     while (isdigit(m_current_char)) {
-        consume();
+        consume(&current_token);
     }
     return true;
 }
 
-bool Lexer::consume_hexadecimal_number()
+bool Lexer::consume_hexadecimal_number(StringBuilder& current_token)
 {
-    consume();
+    consume(&current_token);
+    consume(&current_token);
```

Comment:
Why an extra call to `consume`?

Reply:
Because I'm an idiot and didnt't realize the `0` is consumed in `consume_numeric_literal`. Let me check if there is a test for hex numbers.